docs/concepts/07_dip/05_dapp_developer.md

```diff
@@ -26,8 +26,116 @@ Import the SDK into your code:
+```
+
+:::info What's a base proof?
+A Base proof is a cross chain state proof, representing the parts of a DID stored on the KILT blockchain.
+:::
+
+The configuration takes the following parameters:
+
+```typescript
+
+```
+
+### 2. Generate a submittable extrinsic
+
+The method returns the DID base proof. You have to call a second method, the `[generateDipSubmittableExtrinsic](https://kiltprotocol.github.io/dip-sdk/functions/generateDipSubmittableExtrinsic.html)` method to generate a submittal extrinsic.
+
+You need to pass the following parameters:
+
+-   The API of the consumer chain.
+-   The base proof.
+-   The extrinsic call to the consumer chain.
+-   The DID URI.
+
+```typescript
+const dipSubmittable = DipSdk.generateDipSubmittableExtrinsic({
+    api: consumerApi,
+    baseDipProof,
+    call,
+    didUri: did.uri,
+})
+```
+
+The method returns the different components of the proof:
+
+-   The provider head proof, which is proof of the provider parachain header on the relay chain.
+-   The commitment proof, which proves the deep commitment for a specific subject, which is the DID URI.
+-   The actual deep proof, which reveals parts of the DID document as specified by key IDs, proof version, whether to include the web rename and the linked account, and the proof version that the user specified.
+
+Behind the scenes, the method uses the `dispatchAs` method to pass the extrinsic following the consumer's type registry.
+You can now sign and submit to the consumer chain.
+
+```typescript
+await signAndSubmitTx(consumerApi, dipSubmittable, submitterKeypair)
+```
+
+### 3. Linking accounts (optional)
+
+Linked accounts let you specify which accounts you want to prove that you control when you make the cross-chain proof.
+
+For all the accounts you want to link, use the `associateAccountToChainArgs` method, passing the account and DID to link.
+
+```typescript
+const linkAccountTxs = await Promise.all(
+    linkedAccounts.map(async (acc) => {
+        const functionArgs = await Kilt.Did.associateAccountToChainArgs(
+            acc.address,
+            newFullDidUri,
+            async (input) => acc.sign(input, { withType: true })
+        )
+        return providerApi.tx.didLookup.associateAccount(...functionArgs)
+    })
+)
+```
+
+You can then batch all the linked account transactions and authorize them using the `authorizeTx` method.
+
+```typescript
+const signedLinkedAccounts = await Kilt.Did.authorizeTx(
+    newFullDidUri,
+    providerApi.tx.utility.batchAll(linkAccountTxs),
+    signCallback,
+    newSubmitterKeypair.address as KiltAddress,
+    { txCounter: new BN(4) }
+)
+```
+
+## Creating extensions for specific proofs
+
+If you need a specific proof type for your consumer chain, then you can submit a PR to the SDK repository in the `src > dipProof > extensions` folder.
+The example application includes an extension for a time-bound DID signature. This adds a time-limited DID signature.
```

Comment:
I feel that repeating time-bound DID signature and time-limited DID signature is redundant. I would rather say:

```suggestion
The example application includes an extension for a time-bound DID signature, i.e., a signature which is valid only until a certain block number.
```